rectools/metrics/intersection.py

```diff
@@ -12,7 +12,8 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-from typing import Dict, Hashable, Optional, Union
+from collections.abc import Hashable
```

Comment:
Why do we need this change?

Reply:
This is because `typing.Hashable` will be deprecated, and Python 3.12 warns it https://github.com/python/cpython/issues/94309
It causes lint error on Python 3.12